zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -332,10 +332,22 @@ private void loadFromFile() throws IOException {
       // previously created setting should turn this feature on here.
       setting.getOption().setRemote(true);
 
-      InterpreterSetting intpSetting = new InterpreterSetting(setting.id(), setting.getName(),
+      final InterpreterSetting intpSetting = new InterpreterSetting(setting.id(), setting.getName(),
           setting.getGroup(), setting.getInterpreterInfos(), setting.getProperties(),
           setting.getDependencies(), setting.getOption());
 
+      try {
+        synchronized (interpreterSettings) {
+          loadInterpreterDependencies(intpSetting);
```

Comment:
The issue that i faced when loading interpreters synchronously:
The dependencies can take some time(time to download from mvn , erroneous scenarios etc). This will block loading other interpreters(since it is load method).

But the Notebook websocket server is running. The notebooks web UI will render properly. But when a para is run, the UI will seems to not respond(the para status will not change), because the interpreter might not have got loaded or in the process of being loaded.

It is good to show some UI feedback in these cases.
@prabhjyotsingh